### ISSUE TITLE
TypeScript error when using base Notification class

### ISSUE THREAD

**Issue by bfelbo** [AndrewBarba/apns2#22]
It's amazing to see a well-maintained Node APNS library! The other libraries have been such a mess.

I'm getting a strange TypeScript error when using the base Notification class. The docs specify that for full control, one can create a base Notification like this:
```
const { Notification } = require('apns2')

let notification = new Notification(deviceToken, {
  aps: { ... }
})
```

However, that doesn't seem aligned with the TypeScript types. If I write e.g. 
```
  let notification = new Notification(deviceToken, {
    aps: { alert: { body: "Hey" } }
  })
```
then I get this error:
```
Argument of type '{ aps: { alert: { body: string; }; }; }' is not assignable to parameter of type 'NotificationOptions'.
  Object literal may only specify known properties, and 'aps' does not exist in type 'NotificationOptions'.
```

**Comment by bfelbo** [comment]
Maybe this is related to a lack of specification of `NotificationOptions`?

https://github.com/AndrewBarba/apns2/blob/533643a6a19f17d17fba69bb7e528d58d1a8889d/index.d.ts references `NotificationOptions`, but it's not clear where it's specified. In VS Code, if I click 'Go to Type Definition' for `NotificationOptions`, it opens up something completely unrelated with the following definition:

```
interface NotificationOptions {
    actions?: NotificationAction[];
    badge?: string;
    body?: string;
    data?: any;
    dir?: NotificationDirection;
    icon?: string;
    image?: string;
    lang?: string;
    renotify?: boolean;
    requireInteraction?: boolean;
    silent?: boolean;
    tag?: string;
    timestamp?: number;
    vibrate?: VibratePattern;
}
```

**Comment by AndrewBarba** [comment]
@bfelbo Yup that's definitely it. I think we just need to add `aps` into the `NotificationOptions` interface

**Comment by bfelbo** [comment]
Thanks for looking into this. I think it needs more than adding `aps` as the typing for SilentNotification and BasicNotification also doesn't seem to work. I get errors for:
```
return new SilentNotification(token, { topic, expiration, data })
```
and
```
return new BasicNotification(token, body, { topic, expiration, data, alert })
```

**Comment by AndrewBarba** [comment]
Oh and you're right, it doesn't even seem to be defined. Working on this now give me a few

**Comment by AndrewBarba** [comment]
Can you try v8.1.0? https://github.com/AndrewBarba/apns2/releases/tag/8.1.0

**Comment by AndrewBarba** [comment]
I think it's still missing some things that you need. Let me know if that at least fixed it for `aps` and then I can add the rest

**Comment by bfelbo** [comment]
I still get errors. It seems like `expiration`, `alert` and potentially others are not defined in the type interface?

It'd be nice to have separate type interfaces for the different notification types. Right now it's too easy to supply e.g. `alert` when creating a `SilentNotification`, which would have no effect.

**Comment by AndrewBarba** [comment]
That makes sense

**Comment by AndrewBarba** [comment]
I think this at least covers the correct options:

```typescript
declare interface NotificationOptions {
  alert?: string | any;
  aps?: any;
  badge?: number;
  category?: string;
  collapseId?: string;
  contentAvailable?: boolean;
  data?: any;
  expiration?: number;
  priority?: string;
  pushType?: string;
  sound?: string;
  threadId?: string;
  topic?: string;
}
```

**Comment by bfelbo** [comment]
Great, 8.2 works!

**Comment by bfelbo** [comment]
Getting better types would be amazing when you have a moment. I just realized that this library requires specifying `sound` separately, whereas APNS itself specifies it as part of the `alert` payload, which I'd been doing.

**Comment by AndrewBarba** [comment]
According to apple's docs, sound is not a valid param in the alert payload:

https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW5

But yes I agree with you. Have types that match everything in [here](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW1) would be a huge win

**Comment by bfelbo** [comment]
Oh, my bad!
